File: examples/discovery/discovery.py
from pylebai import zeroconf


def main():
    discovery = zeroconf.Discovery()
    controllers = discovery.resolve()
    print(f"Found {len(controllers)} controller(s)")
    for controller in controllers:
        print(controller.ip_address, controller.model, controller.hostname)


if __name__ == "__main__":
    main()

File: examples/discovery/discovery.cc
#include <iostream>

#include <lebai/discovery.hh>

int main() {
  lebai::zeroconf::Discovery discovery;
  const auto controllers = discovery.resolve();

  std::cout << "Found " << controllers.size() << " controller(s)" << std::endl;
  for (const auto &controller : controllers) {
    std::cout << controller.ip_address << " " << controller.model << " "
              << controller.hostname << std::endl;
  }
  return 0;
}
